pkg/binlog/reader/reader.go

```diff
@@ -42,3 +42,9 @@ type Reader interface {
 	// Status returns the status of the reader.
 	Status() interface{}
 }
+
+// Streamer provides the ability to get binlog event from remote server or local file.
+type Streamer interface {
```

Comment:
can't we use this interface from streamer folder? so we could write

streamer.GetGTIDsForPos and reader.GetGTIDsForPos

Reply:
Do you mean declare this in `streamer` folder? This will cause a circular reference.